### PR TITLE
Feature/vcs gitlab module

### PR DESCRIPTION
**Pull request by alysenko-bekitzur**
### Requirements

* Filling this template is required. Pull request without enough information to be reviewed may be closed

### CODE REVIEW

File: platform/terraform/modules/vcs_gitlab/repository/main.tf [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      # https://registry.terraform.io/providers/gitlabhq/gitlab/latest/docs
+      source  = "gitlabhq/gitlab"
+      version = "16.7.0"
+    }
+  }
+}
+
+
+# Configure Git Provider
+provider "gitlab" {
```
Comment (sergs-pci) [inline platform/terraform/modules/vcs_gitlab/repository/main.tf:14]: Suggest to remove this block. With it I'm receiving 

> │ Warning: Redundant empty provider block
> │
> │   on ../modules/vcs_gitlab/repository/main.tf line 14:
> │   14: provider "gitlab" {
> │
> │ Earlier versions of Terraform used empty provider blocks ("proxy provider configurations") for child modules to declare their need to be passed a provider configuration by their callers. That approach was ambiguous and is
> │ now deprecated.
> │
> │ If you control this module, you can migrate to the new declaration syntax by removing all of the empty provider "gitlab" blocks and then adding or updating an entry like the following to the required_providers block of
> │ module.vcs.module.gitops-repo:
> │     gitlab = {
> │       source = "gitlabhq/gitlab"
> │     }
> │
> │ (and one more similar warning elsewhere)

File: platform/terraform/modules/vcs_gitlab/repository/main.tf [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+# Protect the main branch of the repository. Additionally, require 
+# only allow the engineers team merge to the branch.
+
+resource "gitlab_branch_protection" "this" {
+  count   = var.branch_protection ? 1 : 0
+  project = gitlab_project.repo.id
+
+  branch            = "main"
+  allow_force_push  = false
+  push_access_level = "no one"
```
Comment (sergs-pci) [inline platform/terraform/modules/vcs_gitlab/repository/main.tf:47]: Afraid that can block some automation that we had. Need additionally discuss this

File: platform/terraform/modules/vcs_gitlab/repository/team.tf [hunk 3]
```diff
@@ -0,0 +1,64 @@
+data "gitlab_current_user" "current_user" {}
```
Comment (sergs-pci) [inline platform/terraform/modules/vcs_gitlab/repository/team.tf:1]: Why do we need to have this?

File: platform/terraform/modules/vcs_gitlab/repository/team.tf [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+resource "gitlab_group" "owners" {
```
Comment (sergs-pci) [inline platform/terraform/modules/vcs_gitlab/repository/team.tf:3]: Where is the gitlab_project_share_group resource for the owner group?

File: platform/terraform/modules/vcs_gitlab/repository/variable.tf [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+variable "delete_branch_on_merge" {
+  type    = bool
+  default = true
+}
+
+variable "branch_protection" {
+  type    = bool
+  default = true
```
Comment (sergs-pci) [inline platform/terraform/modules/vcs_gitlab/repository/variable.tf:42]: Not sure that TRUE is suitable for the default value here. Let's discuss.

**Review by sergs-pci — CHANGES_REQUESTED** [review]

File: platform/terraform/modules/vcs_gitlab/repository/team.tf (hunk 3 above)
Comment (alysenko-bekitzur) [reply platform/terraform/modules/vcs_gitlab/repository/team.tf:1]: Removed

File: platform/terraform/modules/vcs_gitlab/repository/team.tf (hunk 4 above)
Comment (alysenko-bekitzur) [reply platform/terraform/modules/vcs_gitlab/repository/team.tf:3]: Added

File: platform/terraform/modules/vcs_gitlab/repository/variable.tf (hunk 5 above)
Comment (alysenko-bekitzur) [reply platform/terraform/modules/vcs_gitlab/repository/variable.tf:42]: Set default to false

**Review by sergs-pci — APPROVED** [review]

**Review by sergs-pci — DISMISSED** [review]

File: tools/cli/services/vcs/github/github_manager.py [hunk 6]
```diff
@@ -85,6 +86,23 @@ def get_current_user_info(self):
...
+        details such as the source and version of the provider.
+
+        Returns:
+            str: A multiline string containing the GitLab provider configuration snippet.
+        """
+        return textwrap.dedent("""\
+        github = {
+              # https://registry.terraform.io/providers/integrations/github/latest/docs
+              source  = "integrations/github"
+              version = "~> 5.42.0"
```
Comment (all4code) [inline tools/cli/services/vcs/github/github_manager.py:103]: Please move version to constants (common/const/const.py)

File: tools/cli/services/vcs/gitlab/gitlab_manager.py [hunk 7]
```diff
@@ -170,7 +174,24 @@ def create_tf_module_snippet(self) -> str:
...
+        details such as the source and version of the provider.
+
+        Returns:
+            str: A multiline string containing the GitLab provider configuration snippet.
+        """
+        return textwrap.dedent("""\
+        gitlab = {
+              # https://registry.terraform.io/providers/gitlabhq/gitlab/latest/docs
+              source  = "gitlabhq/gitlab"
+              version = "16.7.0"
```
Comment (all4code) [inline tools/cli/services/vcs/gitlab/gitlab_manager.py:193]: Please move version to constants (common/const/const.py)

**Review by all4code — CHANGES_REQUESTED** [review]

**Review by all4code — APPROVED** [review]

**Review by sergs-pci — APPROVED** [review]

**Pull request merged**